src/components/autosuggest/_macro.spec.js

```diff
@@ -5,136 +5,258 @@ import * as cheerio from 'cheerio';
+            });
+        });
+
+        describe('WHEN: allowMultiple is false', () => {
+            const $ = cheerio.load(
+                renderComponent('autosuggest', {
+                    ...EXAMPLE_AUTOSUGGEST,
+                    allowMultiple: false,
+                }),
+            );
+            test('THEN: it does not have the data-allow-multiple attribute on the container element', () => {
+                expect($('.ons-autosuggest').attr('data-allow-multiple')).toBeUndefined();
+            });
+        });
+
+        describe('WHEN: allowMultiple is not set', () => {
+            const $ = cheerio.load(
+                renderComponent('autosuggest', {
+                    ...EXAMPLE_AUTOSUGGEST,
+                }),
+            );
+            test('THEN: it does not have the data-allow-multiple attribute', () => {
+                expect($('.ons-autosuggest').attr('data-allow-multiple')).toBeUndefined();
+            });
+        });
     });
 
-    it('has the `data-allow-multiple` attribute when `allowMultiple` is `true`', () => {
-        const $ = cheerio.load(
-            renderComponent('autosuggest', {
-                ...EXAMPLE_AUTOSUGGEST,
-                allowMultiple: true,
-            }),
-        );
+    describe('GIVEN: Params: externalInitialiser', () => {
+        describe('WHEN: externalInitialiser is true', () => {
+            const $ = cheerio.load(
+                renderComponent('autosuggest', {
+                    ...EXAMPLE_AUTOSUGGEST,
+                    externalInitialiser: true,
+                }),
+            );
+
+            test('THEN: it does not have the ons-js-autosuggest class that indicates the component should initialise itself', () => {
+                expect($('.ons-autosuggest').hasClass('ons-js-autosuggest')).toBe(false);
+            });
+        });
+
+        describe('WHEN: externalInitialiser is false', () => {
+            const $ = cheerio.load(
+                renderComponent('autosuggest', {
+                    ...EXAMPLE_AUTOSUGGEST,
+                    externalInitialiser: false,
+                }),
+            );
 
-        expect($('.ons-autosuggest').attr('data-allow-multiple')).toBe('true');
+            test('THEN: it has the ons-js-autosuggest class that indicates the component should initialise itself', () => {
+                expect($('.ons-autosuggest').hasClass('ons-js-autosuggest')).toBe(true);
+            });
+        });
     });
 
-    it('has a special class that indicates the component should initialise itself', () => {
-        const $ = cheerio.load(renderComponent('autosuggest', EXAMPLE_AUTOSUGGEST));
+    describe('GIVEN: Params: isEditable', () => {
+        describe('WHEN: isEditable is false', () => {
+            const $ = cheerio.load(
+                renderComponent('autosuggest', {
+                    ...EXAMPLE_AUTOSUGGEST,
+                    isEditable: false,
```

Comment:
We could add a test for when this is false

Reply:
done